nixos/modules/system/boot/loader/grub/install-grub.pl

```diff
@@ -562,11 +562,18 @@ sub addProfile {
             next;
         }
         my $date = strftime("%F", localtime(lstat($link)->mtime));
-        my $version =
-            -e "$link/nixos-version"
-            ? readFile("$link/nixos-version")
-            : basename((glob(dirname(Cwd::abs_path("$link/kernel")) . "/lib/modules/*"))[0]);
-        addEntry("NixOS - Configuration " . nrFromGen($link) . " ($date - $version)", $link);
+        my $nixos_version = readFile("$link/nixos-version");
+        # Figure out the kernel version
+        my $kernel_dir = dirname(Cwd::abs_path("$link/kernel"));
+        my @module_dirs = glob("$kernel_dir/lib/modules/*");
+        my $kernel_version = "unknown";
+        if (scalar @module_dirs > 0) {
+            $kernel_version = basename($module_dirs[0]);
```

Comment:
We might want to add kernel_version as an additional flag, either opt-in or opt-out, to at least allow for past behavior.
While the regex seems to be working fine I'm wondering if it would be possible to do this from within Nix and what advantages would it bring? My gut feeling would be that moving this from runtime to ben time might be a finicky for very little effort though, so feel free to disregard, my main concern would be about future proofing, or at least make it obvious we depend on this so breakage doesn't happen without noticing. A test is probably a better idea actually!